src/v2/manifest/mod.rs

```diff
@@ -286,28 +286,43 @@ fn evaluate_media_type(
     }
 }
 
-fn build_accept_headers() -> header::HeaderMap {
-    header::HeaderMap::from_iter(
-        [
-            // accept header types and their q value, as documented in
-            // https://tools.ietf.org/html/rfc7231#section-5.3.2
-            (mediatypes::MediaTypes::ManifestV2S2, 0.5),
-            (mediatypes::MediaTypes::ManifestV2S1Signed, 0.4),
-            // TODO(steveeJ): uncomment this when all the Manifest methods work for it
-            // mediatypes::MediaTypes::ManifestList,
-        ]
-        .iter()
-        .filter_map(|(ty, q)| {
-            match header::HeaderValue::from_str(&format!("{}; q={}", ty.to_string(), q)) {
-                Ok(header_value) => Some((header::ACCEPT, header_value)),
-                Err(e) => {
-                    let msg = format!("failed to parse HeaderValue from str: {}:", e);
-                    error!("{}", msg);
-                    None
+fn build_accept_headers(registry: &str) -> header::HeaderMap {
+    // GCR incorrectly parses `q` parameters, so we use special Accept for it.
+    // Bug: https://issuetracker.google.com/issues/159827510.
+    // TODO: when bug is fixed, this workaround should be removed.
+    let no_q = registry.ends_with("gcr.io") || registry.ends_with(".gcr.io");
```

Comment:
```suggestion
    let no_q = registry == "gcr.io" || registry.ends_with(".gcr.io");
```

The previous statement would evaluate to true for strings which you previously intended to exclude in https://github.com/camallo/dkregistry-rs/pull/162#discussion_r446432719, namely `my-super-gcr.io`.

Reply:
Committed and squashed, thanks.